examples/fabric-admin/device_manager/DeviceSynchronization.cpp

```diff
@@ -131,6 +126,12 @@ void DeviceSynchronizer::OnReportEnd()
     if (!DeviceMgr().IsCurrentBridgeDevice(mCurrentDeviceData.node_id))
     {
         AddSynchronizedDevice(mCurrentDeviceData);
+        // TODO(#35077) Figure out how we should reflect CADMIN values of ICD.
+        if (!mCurrentDeviceData.is_icd)
+        {
+            VerifyOrDie(mController);
+            DeviceSubscriptionManager::Instance().StartSubscription(*mController, mCurrentDeviceData.node_id);
```

Comment:
DeviceSynchronizer start the StartSubscription from outside of DeviceSubscriptionManager, but not able to manage it any more.

If the DeviceSubscription fails to clean up properly or "kill itself", it could lead to resource leaks and potentially leave the DeviceSubscriptionManager in an inconsistent state

If we have another DeviceSynchronizer::OnReportEnd(), it seems  DeviceSubscriptionManager::Instance().StartSubscription just be a non-op